### PR TITLE
ci: Add major release tags and risky automation change warning

### PR DESCRIPTION
**Pull request by grayside**
This PR adds 2 GitHub Workflows.

### Workflow 1: major-release-tags

The primary goal of this PR.

When a release is published, the major version will get a dedicated tag (e.g., v0, v1, v2). This allows content that's compatible with a particular major version to follow compatible updates of the codebase without needing to reference more specific release numbers.

If the workflow fails, the tag can be manually created the usual ways.

### Workflow 2: review-warning

Because the major-release-tags workflow requires write permission to the repository, it has a riskier impact if a malicious PR is sent. The review-warning workflow notes PRs that modify the major-release-tags workflow and adds a warning via comment to the PR. We could also have a label, but there's no standard label for "pay extra attention here!"

Screenshot 2023-04-17 at 12 52 05 PM

### CODE REVIEW

**Comment by grayside** [comment]
The lint failure is related to documentation generation needing to be re-run, but that is unrelated to the change in this PR.

**Comment by glasnt** [comment]
Some lint errors cleaned up, but some more popped up. Will address in a separate PR, to try and debug.

**Review by glasnt — APPROVED** [review]

**Comment by glasnt** [comment]
Nope, this lint error is valid: 

```
Error: No newline at end of file ./.github/workflows/release-major-tags.yml
```

File: .github/workflows/release-major-tags.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        if: ${{ steps.parse-major.outputs.result }}
+      - name: delete existing major tag
+        run: |
+          (git tag -d ${{ steps.parse-major.outputs.result }} && git push origin :${{ steps.parse-major.outputs.result }}) || true
+        if: ${{ steps.parse-major.outputs.result }}
+      - name: create major tag
+        run: |
+          git tag ${{ steps.parse-major.outputs.result }} ${{ github.event.GITHUB_REF }}
+          git push origin --tags
+        if: ${{ steps.parse-major.outputs.result }}
```
Comment (glasnt) [inline .github/workflows/release-major-tags.yml:57]: Fix linting error by adding a new line

```suggestion
        if: ${{ steps.parse-major.outputs.result }}

```

**Pull request merged**